main.js

```diff
@@ -0,0 +1,75 @@
+import Producto from './Producto.js';
+import Catalogo from './Catalogo.js';
+import Carrito from './Carrito.js';
+
+// Lista de productos disponibles
+const listaProductos = [
+  new Producto(1, "Alimento para perro", 3500),
+  new Producto(2, "Juguete de goma", 1200),
+  new Producto(3, "Rascador para gatos", 4200),
+  new Producto(4, "Collar con luz", 2800)
+];
+
+// Instancia del catálogo
+const catalogo = new Catalogo(listaProductos);
+
+// Muestra el catálogo ANTES de pedir el número
+function solicitarProductoIndex(productos) {
+  const mensajeCatalogo = catalogo.generarCatalogoTexto();
+  alert("📦 Productos disponibles:\n\n" + mensajeCatalogo);
```

Comment:
No hace falta separarlo, lo podes hacer directamente en el prompt, concatenando correctamente
```suggestion
let mensaje = "📦 Productos disponibles:\n\n";
  productos.forEach((producto, index) => {
    mensaje += `${index + 1}. ${producto.nombre}\n`; // o el campo que uses
  });
  mensaje += "\nIngresá el número del producto que querés agregar:";
  
  const input = prompt(mensaje);```